### ISSUE TITLE
FM

### ISSUE THREAD

**Issue by amauryma55** [amauryma55/python#4]
import joblib

# Define el modelo final (puede ser grid_xgb.best_estimator_, grid_rf.best_estimator_, etc.)
modelo_final = grid_xgb.best_estimator_

# Guarda el modelo en un archivo .pkl
joblib.dump(modelo_final, 'modelo_final.pkl')

import joblib

# Cargar el modelo desde el archivo .pkl
modelo_cargado = joblib.load('modelo_final.pkl')

import pandas as pd
from sklearn.metrics import classification_report, roc_auc_score

# Cargar la base nueva
df_nuevos = pd.read_csv('ruta_a_tu_archivo.csv')

# Separar IDs, explicativas y respuesta real
ids = df_nuevos[Vars_id]
X_nuevos = df_nuevos[Vars_expl]
y_real = df_nuevos[Var_res]

# Calificar usando el modelo cargado
y_pred = modelo_cargado.predict(X_nuevos)
y_prob = modelo_cargado.predict_proba(X_nuevos)[:, 1]

# Evaluación
print('\n=== Resultados Modelo Final ===')
print(classification_report(y_real, y_pred))
print('ROC-AUC:', roc_auc_score(y_real, y_prob))

# Exportar resultado
resultado = pd.concat([ids,
                       pd.Series(y_real, name='var_res_real'),
                       pd.Series(y_pred, name='prediccion'),
                       pd.Series(y_prob, name='probabilidad')],
                      axis=1)

resultado.to_csv('resultado_final_modelo.csv', index=False)
